cloud/amazon/ec2.py

```diff
@@ -1304,7 +1304,9 @@ def startstop_instances(module, ec2, instance_ids, state, instance_tags):
                 except EC2ResponseError, e:
                     module.fail_json(msg='Unable to change state for instance {0}, error: {1}'.format(inst.id, e))
                 changed = True
+            existing_instances_array.append(inst.id)
 
+    instance_ids = list(set(existing_instances_array + (instance_ids or [])))
```

Comment:
Instead of doing instance_ids or [] here, perhaps the instance_ids can default to an empty list in the conditional block at line 1266.  Something like:

``` python
    if instance_ids is None:
        instance_ids = []
    # We need to either have instance_ids or instance_tags defined
    if len(instance_ids) < 1 and not instance_tags:

             module.fail_json(msg='instance_ids should be a list of instances, aborting')
```